### PR TITLE
Add view for admins to create users

### PR DESCRIPTION
**Pull request by anoverga17**
## Motivation and Context

As part of issue #5830, admins should be able to manage users. As part of this, they should be able to create users via the UI. This pull request thus aims to add an admin exclusive view that allows admins to create new end and admin users using the UI. 

## Your Changes

**Description**:
- Added new user route and controller method
- Added new user view
- Added create user route and controller method
- Added link to create user from the users list page
- Added tests for changes

These changes create the following new view:

admin_user_create

**Type of change** (select all that apply):

- [x] New feature (non-breaking change which adds functionality)

## Testing

Manual testing was done through the web interface. As an admin, an Admin and End User were both created through the UI. The user was then checked through the UI and database to ensure it was updated properly and the correct views were being loaded while doing so.
Rspec tests were also added to ensure the users were being updated appropriately.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the pre-commit.ci checks have passed. 
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls. 
- [x] I have added tests for my changes. 
- [x] I have updated the Changelog.md file. 
- [x] I have made changes to the documentation and linked to that pull request below. 

### Pull request to make documentation changes (if applicable)

https://github.com/MarkUsProject/Wiki/pull/102

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -13,6 +13,25 @@ def index
...
     end
 
+    def new
+      @user = User.new
+    end
+
+    def create
+      user_params = params.require(:user)
+      user_type = user_params[:type].camelize.downcase
+      case user_type
```
Comment (anoverga17) [inline app/controllers/admin/users_controller.rb:23]: Unsure if this is the best way to check if the type is valid

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -13,6 +13,27 @@ def index
...
+      when 'enduser'
+        user_params[:type] = 'EndUser'
+      when 'adminuser'
+        user_params[:type] = 'AdminUser'
+      else
+        user_params[:type] = nil
+      end
+      @user = User.create(user_params.permit(*DEFAULT_FIELDS))
+      respond_with @user do |format|
+        format.html { redirect_to admin_users_path }
```
Comment (anoverga17) [inline app/controllers/admin/users_controller.rb:33]: Overriding the response format like this seems to resolve the failing tests. Not sure why.

Comment (anoverga17) [reply app/controllers/admin/users_controller.rb]: Resolved. It took embarrassingly too long, but after leaving and coming back to this problem I finally realize tests failed due to errors with the test itself as opposed to the code.

File: app/controllers/admin/users_controller.rb (hunk 1 above)
Comment (david-yz-liu) [reply app/controllers/admin/users_controller.rb:23]: The `case` is okay, but don't do the `camelize` and `downcase` stuff, just ensure that the input values are the literal names of the classes (`AdminUser`/`EndUser`).

File: spec/controllers/admin/users_controller_spec.rb [hunk 3]
```diff
@@ -98,6 +124,70 @@
...
+          }
+        }
+      end
+      let(:invalid_params) do
+        {
+          user: {
+            user_name: 'notValidUser',
+            email: 'sample@sample.com',
+            id_number: 100_678_901,
+            type: 'Not a valid type',
```
Comment (david-yz-liu) [inline spec/controllers/admin/users_controller_spec.rb:153]: Make this a separate test case (do not combine with other invalid attributes)

File: app/controllers/admin/users_controller.rb [hunk 4]
```diff
@@ -13,6 +13,25 @@ def index
...
+      user_type = user_params[:type].camelize.downcase
+      case user_type
+      when 'enduser'
+        user_params[:type] = 'EndUser'
+      when 'adminuser'
+        user_params[:type] = 'AdminUser'
+      else
+        user_params[:type] = nil
+      end
+      @user = User.create(user_params.permit(*DEFAULT_FIELDS))
```
Comment (david-yz-liu) [inline app/controllers/admin/users_controller.rb:31]: This doesn't need to be an instance variable, you can use `user` instead of `@user`

Comment (anoverga17) [reply app/controllers/admin/users_controller.rb:31]: Removing the `@user` seems to create errors with regards to flash interpolation

**Review by david-yz-liu — APPROVED** [review]

**Pull request merged**